pkg/scheduler/api/validation/validation.go

```diff
@@ -74,20 +69,11 @@ func ValidatePolicy(policy schedulerapi.Policy) error {
 	return utilerrors.NewAggregate(validationErrors)
 }
 
-// validatePriorityRedeclared checks if any custom priorities have been declared multiple times in the policy config
-// by examining the specified priority arguments
-func validatePriorityRedeclared(priorities map[string]schedulerapi.PriorityPolicy, priority schedulerapi.PriorityPolicy) error {
-	var priorityType string
-	if priority.Argument != nil {
-		if priority.Argument.LabelPreference != nil {
-			priorityType = "LabelPreference"
-		} else if priority.Argument.RequestedToCapacityRatioArguments != nil {
-			priorityType = "RequestedToCapacityRatioArguments"
-		} else if priority.Argument.ServiceAntiAffinity != nil {
-			priorityType = "ServiceAntiAffinity"
-		} else {
-			return fmt.Errorf("No priority arguments set for priority %s", priority.Name)
-		}
+// validateRequestedToCapacityRatioRedeclared checks if the RequestedToCapacityRatioRedeclared custom priority has been
+// declared multiple times in the policy config by examining the specified priority arguments
+func validateRequestedToCapacityRatioRedeclared(priorities map[string]schedulerapi.PriorityPolicy, priority schedulerapi.PriorityPolicy) error {
+	if priority.Argument != nil && priority.Argument.RequestedToCapacityRatioArguments != nil {
+		priorityType := "RequestedToCapacityRatio"
```

Comment:
if we're just checking for this one type of priority, there's no need to declare this as a variable

Reply:
actually, if we're just checking for the one type this whole function could be moved to the main loop in `ValidatePolicy`, and you wouldn't need a map to store previously-declared priorities, you could just have one variable that you check against to see if it's been set

Reply:
Just to avoid typing the string multiple times.

Reply:
Twice? The variable served a functional purpose when there were multiple different cases for its value which changed what we searched for in the map. It's a small nit but just a suggestion and something to think about

Reply:
Removed variable.